app/models/labware_creators/plate_split_to_tube_racks.rb

```diff
@@ -0,0 +1,453 @@
+    #
+    # returns [Array of hashes] e.g.
+    # [
+    #   {
+    #     prefix: SPARE,
+    #     name: NT11111111:A1,
+    #     foreign_barcode: FD11111111
+    #   },
+    #   etc.
+    # ]
+    #
+    # rubocop:disable Metrics/AbcSize
+    # rubocop:disable Metrics/MethodLength
+    def generate_sequencing_tube_attributes
+      # fetch the available tube positions (i.e. locations of scanned tubes for which we
+      # have the barcodes) e.g. ["A1", "B1", "D1"]
+      available_tube_positions = sequencing_csv_file.position_details.keys
+
+      # used for building request hash later
+      @sequencing_wells_to_tube_posns = {}
+
+      parent_wells_for_sequencing
+        .zip(available_tube_positions)
+        .map do |well, tube_posn|
+          sample_uuid = well.aliquots.first.sample.uuid
+
+          # TODO: put prefixes in the purpose config
+          name_for_details = name_for_details_hash('SEQ', ancestor_tube_barcode(sample_uuid), tube_posn)
+
+          tube_name = name_for(name_for_details)
+          @sequencing_wells_to_tube_posns[well] = tube_name
+
+          { name: tube_name, foreign_barcode: sequencing_csv_file.position_details[tube_posn]['barcode'] }
+        end
+    end
+
+    # rubocop:enable Metrics/AbcSize
+    # rubocop:enable Metrics/MethodLength
+
+    def contingency_tube_attributes
+      @contingency_tube_attributes ||= generate_contingency_tube_attributes
+    end
+
+    #
+    # Create the tube attributes to send for the tubes creation in Sequencescape.
+    # Passes the name for each tube.
+    # Passes the foreign barcode extracted from the tube rack scan upload for each tube,
+    # which on the Sequencescape side sets that barcode as the primary.
+    #
+    # returns [Array of hashes] e.g.
+    # [
+    #   {
+    #     prefix: SPARE,
+    #     name: NT11111111:A1,
+    #     foreign_barcode: FD11111111
+    #   },
+    #   etc.
+    # ]
+    #
+    # rubocop:disable Metrics/AbcSize
+    # rubocop:disable Metrics/MethodLength
+    def generate_contingency_tube_attributes
+      # fetch the available tube positions (i.e. locations of scanned tubes for which we
+      # have the barcodes) e.g. ["A1", "B1", "D1"]
+      available_tube_positions = contingency_csv_file.position_details.keys
+
+      # used for building request hash later
+      @contingency_wells_to_tube_posns = {}
+
+      parent_wells_for_contingency
+        .zip(available_tube_positions)
+        .map do |well, tube_posn|
+          sample_uuid = well.aliquots.first.sample.uuid
+
+          # TODO: put prefixes in the purpose config
+          name_for_details = name_for_details_hash('SPR', ancestor_tube_barcode(sample_uuid), tube_posn)
+
+          tube_name = name_for(name_for_details)
+          @contingency_wells_to_tube_posns[well] = tube_name
+
+          { name: tube_name, foreign_barcode: contingency_csv_file.position_details[tube_posn]['barcode'] }
+        end
+    end
```

Comment:
Seems like a fair bit of code duplication between `generate_sequencing_tube_attributes` and `generate_contingency_tube_attributes` - could probably do with a refactor.

Reply:
refactored